### PR TITLE
Refactor sncast_std

### LINKED ISSUE #1325: Refactor cast_std
https://github.com/foundry-rs/starknet-foundry/pull/1150/files#r1400898991

### PR DESCRIPTION
**Pull request by drknzz**
Closes #1325 

## Introduced changes

-

## Checklist

- [x] Linked relevant issue
- [x] Updated relevant documentation
- [x] Added relevant tests
- [x] Performed self-review of the code
- [x] Added changes to `CHANGELOG.md`

### CODE REVIEW

File: sncast_std/src/lib.cairo [hunk 1]
```diff
@@ -195,3 +137,12 @@ fn get_nonce(block_tag: felt252) -> felt252 {
...
 }
+
+fn extend_array(ref array: Array<felt252>, mut span: Span<felt252>) {
+    loop {
+        match span.pop_front() {
+            Option::Some(x) => { array.append(x.clone()); },
+            Option::None => { break; }
+        };
+    };
+}
```
Comment (Arcticae) [inline sncast_std/src/lib.cairo:148]: There is a function for that since 2.4.0

**Review by Arcticae — CHANGES_REQUESTED** [review]

File: sncast_std/src/lib.cairo [hunk 2]
```diff
@@ -195,3 +137,12 @@ fn get_nonce(block_tag: felt252) -> felt252 {
     let buf = cheatcode::<'get_nonce'>(inputs.span());
     *buf[0]
 }
+
+fn extend_array(ref array: Array<felt252>, mut span: Span<felt252>) {
```
Comment (cptartur) [inline sncast_std/src/lib.cairo:141]: There is a function.

**Review by cptartur — CHANGES_REQUESTED** [review]
There is a function

File: sncast_std/src/lib.cairo (hunk 2 above)
Comment (drknzz) [reply sncast_std/src/lib.cairo]: thanks

Comment (cptartur) [reply sncast_std/src/lib.cairo]: You are welcome.

**Review by cptartur — APPROVED** [review]

**Review by Arcticae — APPROVED** [review]

**Pull request merged**